### PR TITLE
Add SBOM task to release pipeline

### PR DESCRIPTION
**Pull request by davidmrdavid**
Includes SBOM-generation task to our pipeline, which documents our dependencies for cybersecurity compliance, It also uses the new 1ES-approved VM, another compliance requirement.

### CODE REVIEW

File: azure-pipelines-release.yml [hunk 1]
```diff
@@ -58,6 +58,12 @@ steps:
           }
       ]
 
+- task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+  displayName: 'SBOM Generation Task'
+  inputs:
+    BuildDropPath: '$(Build.ArtifactStagingDirectory)'
```
Comment (bachuv) [inline azure-pipelines-release.yml:64]: Should `$(Build.ArtifactStagingDirectory)` be `$(build.artifactStagingDirectory)` to match other places in this yml file or is it fine?

Comment (davidmrdavid) [reply azure-pipelines-release.yml:64]: @bachuv, I don't see `$(build.artifactStagingDirectory` being used in other places. What line(s) are you referring to?

Comment (bachuv) [reply azure-pipelines-release.yml:66]: - https://github.com/Azure/durabletask/blob/a2aa3f0430d3a2ec7a667dc1ca88ae6b1aeb3e04/azure-pipelines-release.yml#L76
- https://github.com/Azure/durabletask/blob/a2aa3f0430d3a2ec7a667dc1ca88ae6b1aeb3e04/azure-pipelines-release.yml#L85
- https://github.com/Azure/durabletask/blob/a2aa3f0430d3a2ec7a667dc1ca88ae6b1aeb3e04/azure-pipelines-release.yml#L107

Comment (davidmrdavid) [reply azure-pipelines-release.yml]: Thanks! My latest commit utilizes this capitalization scheme

File: azure-pipelines-release.yml [hunk 2]
```diff
@@ -58,6 +60,12 @@ steps:
           }
       ]
 
+- task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
```
Comment (bachuv) [inline azure-pipelines-release.yml:63]: Is this `task` value standard for Manifest Generator tasks? When I try populating this task in DevOps, I get `AzureArtifacts.drop-validator-task.manifest-generator-task.ManifestGeneratorTask@0`

Comment (davidmrdavid) [reply azure-pipelines-release.yml:63]: Yeah I think in DevOps there's somehow two ManifestGenerator tasks, one of them resolves to this path, and the other resolves to the path you shared. Ultimately they both seem to do the same thing, so I can't really judge which one is more appropiate.

File: azure-pipelines-release.yml [hunk 3]
```diff
@@ -58,6 +60,12 @@ steps:
...
+  displayName: 'SBOM Generation Task'
```
Comment (bachuv) [inline azure-pipelines-release.yml:64]: Can you explain how we know that this is the correct place to add the SBOM Generation Task? I was looking at another PR: https://github.com/Azure/azure-functions-eventgrid-extension/pull/98/files and it's the last step before publishing the pipeline artifact.

Comment (davidmrdavid) [reply azure-pipelines-release.yml:64]: I thought we had to generate the SBOM before running `pack`, but it seems other PRs disagree. Let me play with the location of this step. If I can still generate the SBOM when running this later in the pipeline, then I'm ok with moving this further down.

Comment (davidmrdavid) [reply azure-pipelines-release.yml]: My latest commit moves this further down

**Review by amdeel — APPROVED** [review]
LGTM!

**Pull request merged**